docs/smart-contracts/guides/conventions/work-contractspec-js.mdx

```diff
@@ -4,4 +4,696 @@ hide_table_of_contents: true
+
+Create a new file called `TokenManagerContract.ts` and define the contract client class using the extracted contract specification.
+
+```typescript
+// TokenManagerContract.ts
+import { Address, Contract } from "@stellar/stellar-sdk";
+import { Spec as ContractSpec, i128 } from "@stellar/stellar-sdk/contract";
+// Contract.Spec used to be ContractSpec so if you're trying to update old code to the new sdk you can just`import { Spec as ContractSpec } from "@stellar/stellar-sdk/contract"; ` as opposed to import { ContractSpec } from "@stellar/stellar-sdk";`
+// Copy the contract spec from the generated TypeScript bindings
+const contractSpec = new ContractSpec([
+  "AAAAAAAAAAAAAAAKaW5pdGlhbGl6ZQAAAAAABAAAAAAAAAAFYWRtaW4AAAAAAAATAAAAAAAAAAJkZXBvc2l0AAAAAAATAAAAAAAAAAl1c2VyAAAAEwAAAAAAAAAGYW1vdW50AAAAAAATAAAAAAAAAAR3aXRoZHJhdwAAAAAABgAAAAAAAAAFcmVzdWx0AAAAAAA=",
+  "AAAAAAAAAAAAAAANZ2V0X2JhbGFuY2UAAAAAAAIAAAAAAAAAAl9yZXN1bHQAAAAAEwAAAAAAAAAJdXNlcl9hZGRyAAAAAAAT",
+  "AAAAAAAAAAAAAAAAAUNvbnRyYWN0RXJyb3JNYW5hZ2VyAAAACwAAAAAAAAAAAAALAA==",
+]);
+
+export class TokenManagerContract extends Contract {
+  static spec = contractSpec;
+
+  constructor(address: string) {
+    super(address);
+  }
+
+  public initialize(admin: Address) {
+    const invokeArgs = TokenManagerContract.spec.funcArgsToScVals(
+      "initialize",
+      {
+        admin,
+      },
+    );
+    const operation = this.call("initialize", ...invokeArgs);
+    return operation.toXDR("base64");
+  }
+
+  public deposit(user: Address, amount: i128) {
+    const invokeArgs = TokenManagerContract.spec.funcArgsToScVals("deposit", {
+      user,
+      amount,
+    });
+    const operation = this.call("deposit", ...invokeArgs);
+    return operation.toXDR("base64");
+  }
+
+  public withdraw(user: Address, amount: i128) {
+    const invokeArgs = TokenManagerContract.spec.funcArgsToScVals("withdraw", {
+      user,
+      amount,
+    });
+    const operation = this.call("withdraw", ...invokeArgs);
+    return operation.toXDR("base64");
+  }
+
+  public get_balance(user: Address) {
+    const invokeArgs = TokenManagerContract.spec.funcArgsToScVals(
+      "get_balance",
+      {
+        user,
+      },
+    );
+    const operation = this.call("get_balance", ...invokeArgs);
+    return operation.toXDR("base64");
+  }
+
+  static readonly parsers = {
+    initialize: (result: string) =>
+      TokenManagerContract.spec.funcResToNative("initialize", result),
+    deposit: (result: string) =>
+      TokenManagerContract.spec.funcResToNative("deposit", result),
+    withdraw: (result: string) =>
+      TokenManagerContract.spec.funcResToNative("withdraw", result),
+    get_balance: (result: string) =>
+      TokenManagerContract.spec.funcResToNative("get_balance", result),
+  };
+}
+
+export const Errors = {
+  1: { message: "Internal error" },
+  2: { message: "Unauthorized error" },
+  3: { message: "Insufficient balance" },
+  4: { message: "Overflow error" },
+};
+```
```

Comment:
It'd be good to mention here that I'm pretty sure that this is what the `Client` does (cc @chadoh?) when you use the bindings!